.github/workflows/nix-check.yml

```diff
@@ -0,0 +1,35 @@
+name: Nix Flake Check
+
+on:
+  pull_request:
+  push:
+    branches:
+      - master
+      - 'release-*'
+      - 'dev-*'
+permissions:
+  contents: read
+
+jobs:
+  flake-check:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v4
+
+      - name: Install Nix
+        uses: DeterminateSystems/nix-installer-action@v9
+
+      - name: Enable flakes
+        uses: DeterminateSystems/magic-nix-cache-action@v2
+
+      - name: Check Nix flake inputs
+        run: nix flake check
+
+      - name: Show flake metadata
+        if: always() # Run even if previous step fails
+        run: nix flake metadata
+
+      - name: Validate flake outputs
+        if: always() # Run even if previous step fails
+        run: nix flake show
```

Comment:
should we have steps for `yarn`, `yarn build`, and `cd packages/cosmic-swingset; make`?

Reply:
when I try to `make` in cosmic-swingset, I get:

```
gyp info spawn args ]
Traceback (most recent call last):
  File "/tmp/agoric-sdk/node_modules/node-gyp/gyp/gyp_main.py", line 42, in <module>
    import gyp  # noqa: E402
    ^^^^^^^^^^
  File "/tmp/agoric-sdk/node_modules/node-gyp/gyp/pylib/gyp/__init__.py", line 9, in <module>
    import gyp.input
  File "/tmp/agoric-sdk/node_modules/node-gyp/gyp/pylib/gyp/input.py", line 19, in <module>
    from distutils.version import StrictVersion
ModuleNotFoundError: No module named 'distutils'
```

Reply:
same symptoms from `./bin/agd build --force` from the top.

Reply:
ok I will look into this

Reply:
your fix works